Service/Request/Visitor/RemoveNullFieldsRequestVisitor.php

```diff
@@ -0,0 +1,96 @@
+<?php
+declare(strict_types=1);
+
+namespace Auto1\ServiceAPIClientBundle\Service\Request\Visitor;
+
+use Http\Message\StreamFactory;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * Class RemoveNullFieldsRequestVisitor.
+ */
+class RemoveNullFieldsRequestVisitor implements RequestVisitorInterface
+{
+    /**
+     * @var EndpointConfiguration[]
+     */
+    private $endpoints;
+
+    /**
+     * @var StreamFactory
+     */
+    private $factory;
+
+    /**
+     * RemoveNullFieldsRequestVisitor constructor.
+     *
+     * @param StreamFactory $factory
+     * @param EndpointConfiguration[] $endpoints
+     */
+    public function __construct(
+        StreamFactory $factory,
+        array $endpoints
+    ) {
+        $this->factory = $factory;
+        $this->endpoints = $endpoints;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function visit(RequestInterface $request): RequestInterface
+    {
+        if ($this->isCompatible($request)) {
+            $request = $request->withBody($this->cleanBody($request->getBody()));
+        }
+
+        return $request;
+    }
+
+    /**
+     * @return bool
+     */
+    private function isCompatible(RequestInterface $request): bool
+    {
+        foreach ($this->endpoints as $endpoint) {
+            if (
+                $request->getMethod() === $endpoint->getMethod()
```

Comment:
I think, you should also check that you will process `json` stream.

Reply:
And if the is not a json, should it throw an exception?